From d5a4f5dd258453a651065e823fe2257d8b309684 Mon Sep 17 00:00:00 2001
From: Hans Leidekker <hans@codeweavers.com>
Date: Fri, 27 Jun 2025 10:23:49 +0200
Subject: [PATCH 284/293] widl: Store EventRegistrationToken reference in the
 real type.

So it can be used to build the method signature.
---
 tools/widl/metadata.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tools/widl/metadata.c b/tools/widl/metadata.c
index 43488548561..3f87bb6762f 100644
--- a/tools/widl/metadata.c
+++ b/tools/widl/metadata.c
@@ -2087,12 +2087,13 @@ static void add_method_params_step1( var_list_t *arg_list )
         type_t *type = arg->declspec.type;
 
         if (type_get_type( type ) == TYPE_POINTER) type = type_pointer_get_ref_type( type );
-
         if (type->name && !strcmp( type->name, "EventRegistrationToken" ))
         {
             UINT assemblyref, scope;
+
             assemblyref = add_assemblyref_row( 0x200, 0, add_string("Windows.Foundation") );
             scope = resolution_scope( TABLE_ASSEMBLYREF, assemblyref );
+            type = type_get_real_type( type );
             type->md.ref = add_typeref_row( scope, add_string("EventRegistrationToken"), add_string("Windows.Foundation") );
         }
     }
-- 
2.43.0

